src/random_test.cpp

```diff
@@ -3228,6 +3228,25 @@ void Thd1::run_some_query() {
       current = table_started++;
     }
 
+    if (options->at(Option::CHECK_TABLE_PRELOAD)->getBool()) {
+      std::cout << "Checking tables preload from thread " << thread_id << std::endl;
+      unsigned long int current = 0;
+      while(current < all_tables->size()) {
+      auto table = all_tables->at(current);
+      if (table->type == Table::PARTITION && rand_int(4) == 1) {
```

Comment:
Our goal is to validate all tables so, better to check all partitions.

Reply:
Move it to a function.
Utilize all threads to check tables. 
Execute it if step<>1

Reply:
Done.